src/components/AgentRow.tsx

```diff
@@ -13,19 +15,67 @@ const AgentRow: React.FC<AgentRowProps> = (
 ): JSX.Element => {
   const { agent } = props;
 
+  const [showChildren, setShowChildren] = React.useState(false);
+  const hasChildren = agent.displayStates.length > 0;
+
+  const getChildRows = (agent: UIDisplayEntry) => {
+    return agent.displayStates.map((displayState) => {
+      const checkboxProps: ChildAgentProps = {
+        agentLevel: AgentLevel.ChildAgent,
+        agentName: agent.name,
+        childName: displayState.name,
```

Comment:
this feels like the level where all the 'is selected' stuff should be happening. you could even define the handler here. That way the props sent into a checkbox could jsut be its value/name, whether it's checked or not, and an onClick handler